sdk/typescript/src/types/objects.ts

```diff
@@ -36,6 +36,21 @@ export type SuiMoveObject = {
   has_public_transfer?: boolean;
 };
 
+export type CoinDenominationInfoResponse = {
+  /** Coin type like "0x2::sui::SUI" */
+  coin_type: string;
+  /** min unit, like MIST */
+  basic_unit: string;
+  /** commonly used unit, like SUI */
+  canonical_unit: string;
+  /** value of 1 canonical over 1 basic unit,
+   * for example 1_000_000_000 = 1 SUI / 1 MIST */
+  denomination: bigint;
```

Comment:
Prefer to remove this to avoid duplication and inconsistency with the `decimal_number`. The user can easily derive `denomination` with `Math.pow`